testcases/InstallUpstreamKernel.py

```diff
@@ -182,3 +191,24 @@ def is_url(path):
         finally:
             if self.console_thread.isAlive():
                 self.console_thread.console_terminate()
+
```

Comment:
@PraveenPenguin @abdhaleegit As this could be a common function we can have in the utility file, For time being to test in CI runs i have added here. As i am not able to recreate the issue i have not tested with real failure. But i have tested the code and installation was successful, This code doesn't break anything in InstallUpstream.py file